Marsiple/App/Todos/TodosViewController.swift

```diff
@@ -42,15 +42,18 @@ extension TodosViewController: UITableViewDataSource {
 
 extension TodosViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        var removedTodo = todos[indexPath.section].remove(at: indexPath.row)
-        removedTodo.completed = !removedTodo.completed
-        todos[1 - indexPath.section].insert(removedTodo, at: todos[1 - indexPath.section].count)
-        tableView.reloadData()
+        let finishIndexPath = IndexPath(row: todos[1 - indexPath.section].count, section: 1 - indexPath.section)
+        tableView.performBatchUpdates({
+            var removedTodo = todos[indexPath.section].remove(at: indexPath.row)
+            removedTodo.completed = !removedTodo.completed
+            todos[finishIndexPath.section].insert(removedTodo, at: finishIndexPath.row)
```

Comment:
Lines `47`, `48`, `49` should not reside in the `performBatchUpdates` method - only `UITableView` mutating methods should reside within a `performBatchUpdates` call.